Etiquette/CODE_REVIEW.md

```diff
@@ -0,0 +1,110 @@
+* Ask, don't tell. ("What do you think about...?", "How about ...?", "Should we try ...?")
+* Be brief but clear in your comments (code review is not about proving who is more correct or knows more)
+* Unless explicitly asked for it do not suggest code to be written in a specific way - let the author to find or suggest a solution themselves, they may come up with a better one
+* Submit all the comments as a single review, even if there is just one comment
+* If you see a stale pull request without any updates - ask for the status update in the comment
+* Do not be a blocker - if you requested changes and they were addressed make sure you review it
+* Do not merge others pull requests - they may be waiting for someone else review. In general, the PR author should be the one adding the "Merge" label to merge their own PRs (except rare cases like if the author is on holidays but the PR is otherwise good to go)
+
+Responding to feedback
+-------------------------
+
+* Wait before answering to a single comment, the reviewer may be still writing more of them (though they should prefer to send all comments at once)
+* Be grateful for the reviewer's suggestions. ("Good call. I'll make that
+  change.")
+* If you disagree with a suggestion, explain why and avoid template answers like "out of the scope" or "thank you for your opinion". If you think that suggestion is a personal preference and does not bring value - say so
+* Assume the best intention from the reviewer's comments.
+* If a reviewer seems aggressive or angry or otherwise personal, consider if it is intended to be read that way and ask the person for clarification of intent, in person if possible. Do not escalate online discussions.
+* If you can't reach agreement seek for more opinions
+* Don't be too defensive about your code, someone will rewrite it anyway sooner or later (it might be yourself as well)
+* Try to answer all comments, don't make your colleagues feel like you are ignoring them. You can review your own code for that instead of sending individual replies so that all your answers are posted at once.
+* Add "work in progress" label while addressing review comments
+* Re-request review when you addressed the comments (either implementing suggested change or replying to the comment) and label it with "Ready for review" again
+* Do not rebase and force-push to your pull request branch after you have received any comments, use merge instead. This allows reviewers to view changes on the PR since their last review, rather than re-reviewing a lot of the code they have already reviewed. All pull requests are squashed when they are merged into the base branch so the history of your changes won't matter. It's fine to rebase and force-push if there are no comments.
+* Do not dismiss any pending requests for changes unless you agreed with their authors that it was addressed
```

Comment:
In this case shouldn't the author mention in a comment that the request has been discussed with the reviewer and the necessary action has been taken?

Reply:
yep, and github actually enforced to enter the reason for dismissal and adds it as a comment, will add this

Reply:
Ahh great! Didn't know that 😄